### PR TITLE
Module to enable Data Lifecycle Manager to take snapshots of a EBS

### LINKED ISSUE #196: Create module to enable Data Lifecycle Manager to take snapshots of a EBS
The Data Lifecycle Manager policy creates snapshots of an EBS that later could be used as backups or to restore data for testing.

# Requirements
- Create a module with AWS Data Lifecycle Manager
- Parametrize the configs

### PR DESCRIPTION
**Pull request by lpaulmp**
Closes #196

Output
This example was deployed in a dev environment.

Screen Shot 2019-05-16 at 19 42 23

### CODE REVIEW

**Comment by lpaulmp** [comment]
@ketzacoatl I added the variable `name_prefix` to generate the schedule name of the DLM policy.

File: modules/dlm-lifecycle-policy/main.tf [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+*
+* The module supports:
+*
+* * Generate snapshots of volumes attached to ec2 instances
+* * Assume the IAM Role policy to manage DLM lifecycle policy
+*
+*/
+
+# DLM lifecycle schedule
+resource "aws_dlm_lifecycle_policy" "gitlab-ebs-lifecycle-policy" {
```
Comment (lpaulmp) [inline modules/dlm-lifecycle-policy/main.tf]: TODO: update the name of resource! for more one more generic

**Comment by lpaulmp** [comment]
@ketzacoatl This MR is ready to merge after upgrading the TF version and adding types for the variables the modules creates the snapshots smoothly, I already tested with our infra.

Also, I made a relevant change for the PR moving the IAM role `modules/dlm-lifecycle-iam-role` creation to another module with this change the  DML policy is clean and does one thing, create the schedule for the snapshot.

File: modules/dlm-lifecycle-iam-role/main.tf [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      "Principal": {
+        "Service": "dlm.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
```
Comment (ketzacoatl) [inline modules/dlm-lifecycle-iam-role/main.tf]: We can merge this as-is, but we should update this code to use the data source so we can use HCL and don't have to do the messy JSON thing.

File: modules/dlm-lifecycle-policy/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+This module creates the policy that manage the creation of EBS snapshots through AWS Data Lifecycle Manager, the policy let you manage the schedule of the snapshot as well as the number of snapshots.
+
+### Example how to use
+
+Define variables
+
+```
+module "ebs-backup-policy" {
+  source = "github.com/fpco/terraform-aws-foundation//modules/dlm-lifecycle-policy"
```
Comment (ketzacoatl) [inline modules/dlm-lifecycle-policy/README.md]: This example should source the module from the registry (we need a ticket for making sure all of our examples and docs do that - #268).

**Pull request merged**